protocol/contracts/v2/strategies/SkyFarmBaseStrategy.sol

```diff
@@ -0,0 +1,227 @@
+pragma solidity 0.8.20;
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Temple (v2/strategies/SkyFarmBaseStrategy.sol)
+
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { IERC4626 } from "@openzeppelin/contracts/interfaces/IERC4626.sol";
+
+import { ITempleBaseStrategy } from "contracts/interfaces/v2/strategies/ITempleBaseStrategy.sol";
+import { AbstractStrategy } from "contracts/v2/strategies/AbstractStrategy.sol";
+import { CommonEventsAndErrors } from "contracts/common/CommonEventsAndErrors.sol";
+
+interface DaiUsds {
+    function daiToUsds(address usr, uint256 wad) external;
+    function usdsToDai(address usr, uint256 wad) external;
+    function dai() external view returns (IERC20);
+    function usds() external view returns (IERC20);
+}
+
+/**
+ * @title DAI to Origami SKY Farm - Base Strategy
+ * @notice Deposit idle DAI from the Treasury Reserve Vaults into the Origami sUSDS+s ERC4626 vault
+ */
+contract SkyFarmBaseStrategy is AbstractStrategy, ITempleBaseStrategy {
+    using SafeERC20 for IERC20;
+
+    string private constant VERSION = "1.0.0";
+
+    /// @notice DAI
+    IERC20 public immutable daiToken;
+
+    /// @notice USDS
+    IERC20 public immutable usdsToken;
+
+    /// @notice The ERC4626 vault token taking USDS deposits
+    IERC4626 public immutable usdsVaultToken;
+
+    /// @notice Maker/Sky provided contract to convert DAI<-->USDS 1:1
+    DaiUsds public immutable daiToUsds;
+
+    event DaiDeposited(uint256 amount);
```

Comment:
I see you don't have an interface file for this strategy yet. Perhaps a future todo?

Reply:
None of the strategies have interfaces as it's not expected to be used. There's a base interface for external clients to use for integration, eg [ITempleStrategy.sol](https://github.com/TempleDAO/temple/blob/stage/protocol/contracts/interfaces/v2/strategies/ITempleStrategy.sol)